src/index.js

```diff
@@ -117,22 +128,53 @@ function check(autoCheckElement: AutoCheckElement) {
     input.setCustomValidity('Verifying…')
   }
   autoCheckElement.dispatchEvent(new CustomEvent('loadstart'))
-  performCheck(input, body, src)
+
+  const options: RequestOptions = {body, method: 'POST'}
+
+  // If there is a controller, it means we are already in flight.
+  // Cancel that request and create a new signal.
+  let controller = abortControllers.get(autoCheckElement)
+  if (controller) {
+    // Cancel the in-flight request.
+    controller.abort()
+  }
+
+  // We need to create a new controller so we can get a new signal?
+  controller = new AbortController()
+  abortControllers.set(autoCheckElement, controller)
+
+  // Set the component as being in-flight
+  options.signal = controller.signal
+
+  fetch(src, options)
+    .then(response => {
+      if (response.status !== 200) {
+        throw new ErrorWithResponse(response.statusText, response)
+      }
+      return response.text()
+    })
     .then(message => {
       autoCheckElement.dispatchEvent(new CustomEvent('load'))
       if (autoCheckElement.required) {
         input.setCustomValidity('')
       }
       input.dispatchEvent(new CustomEvent('auto-check-success', {detail: {message}, bubbles: true}))
+
+      // Mark the component as not being in-flight any more.
+      abortControllers.delete(autoCheckElement)
     })
-    .catch(error => {
+    .catch(async error => {
```

Comment:
An `AbortError` is raised when the controller aborts the fetch, which is caught by this block. So we need to ignore aborted fetches here with a [conditional branch](https://developers.google.com/web/updates/2017/09/abortable-fetch#reacting_to_an_aborted_fetch) testing `error.name === 'AbortError'`.